appengine/go11x/tasks/create_task.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2018 Google LLC. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// [START cloud_tasks_appengine_create_task]
+
+// Package tasks contains snippets for using Cloud Tasks with App Engine.
+package tasks
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	cloudtasks "cloud.google.com/go/cloudtasks/apiv2beta3"
+	taskspb "google.golang.org/genproto/googleapis/cloud/tasks/v2beta3"
+)
+
+// CreateTask creates a new task in your App Engine queue.
+func CreateTask(locationID, queueID, message string) (*taskspb.Task, error) {
+	// Create a new Cloud Tasks client instance.
+	// See https://godoc.org/cloud.google.com/go/cloudtasks/apiv2beta3
+	ctx := context.Background()
+	client, err := cloudtasks.NewClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("NewClient: %v", err)
+	}
+
+	// Build the Task queue path.
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	queuePath := fmt.Sprintf("projects/%s/locations/%s/queues/%s", projectID, locationID, queueID)
+
+	// Build the Task payload.
+	// https://godoc.org/google.golang.org/genproto/googleapis/cloud/tasks/v2beta3#CreateTaskRequest
+	req := &taskspb.CreateTaskRequest{
+		Parent: queuePath,
+		Task: &taskspb.Task{
+			// https://godoc.org/google.golang.org/genproto/googleapis/cloud/tasks/v2beta3#AppEngineHttpRequest
+			PayloadType: &taskspb.Task_AppEngineHttpRequest{
+				AppEngineHttpRequest: &taskspb.AppEngineHttpRequest{
+					HttpMethod:  taskspb.HttpMethod_POST,
+					RelativeUri: "/task_handler",
+				},
+			},
+		},
+	}
+
+	// Add a payload message if one is present.
```

Comment:
Seems like this could be inline? If `message == ""`, setting `Body = []byte("")` should be the same as the default value. I could be wrong.

Reply:
I'm trying to demonstrate how to modify the body payload based on logic here. Figuring out how to do this was the most difficult part of this sample.

Is this something addressable via comment?

```
- // Add a payload message if one is present.
+ // Add a payload based on conditional logic.
```

Trying to think if I can come up with an arbitrary but reasonable conditional without a more specific use case.

Reply:
@tbpg do you have thoughts on this?

Reply:
Ah, gotcha. Protos are tricky for that. In that case, this looks good to me.